controllers/snapshot/snapshot_adapter.go

```diff
@@ -88,6 +88,26 @@ func (a *Adapter) EnsureAllIntegrationTestPipelinesExist() (results.OperationRes
 	return results.ContinueProcessing()
 }
 
+// EnsureGlobalComponentImageUpdated is an operation that ensure the ContainerImage in the Global Candidate List
+// being updated when the ApplicationSnapshot passed all the integration tests
+func (a *Adapter) EnsureGlobalComponentImageUpdated() (results.OperationResult, error) {
+	if gitops.HaveHACBSTestsSucceeded(a.snapshot) {
+		patch := client.MergeFrom(a.component.DeepCopy())
+		for _, component := range a.snapshot.Spec.Components {
+			if component.Name == a.component.Name {
+				a.component.Spec.ContainerImage = component.ContainerImage
+				break
+			}
+		}
+		err := a.client.Patch(a.context, a.component, patch)
```

Comment:
I believe that in order to patch the `Status` subresource we need to use the `client.Status.Patch`, e.g. in this case:
`err := a.client.Status.Patch(a.context, a.component, patch)`

Reply:
Disregard this, I've gotten confirmation that only the component `Spec` is supposed to be updated.